Pycloud Deployer/infra_classes.py

```diff
@@ -0,0 +1,629 @@
+        """Initiate the Server class properties
+
+        Args:
+            token_credential (DefaultAzureCredential): Instance of the DefaultAzureCredential class
+        """
+        self.net_client = NetworkManagementClient(token_credential, subscription_id)
+        self.compute_client = ComputeManagementClient(token_credential, subscription_id)
+
+    def create_webserver(self, group_name, vnet_name, subnet_name,
+                         webserver_name, server_number=0):
+        """Create the webserver instance & its dependencies
+
+        Args:
+            group_name (str): Name of the resource group
+            webserver_name (str): Prefix for webserver's name
+            vnet_name (str): Name of the VNet
+            subnet_name (str): Name of the subnet
+            server_number (int, optional): Suffix for webserver's name. Defaults to 0.
+        """
+        webserver_name_suffix = f"{webserver_name}{server_number}"
+        publicip_name = f"{webserver_name_suffix}_ip"
+        interface_name = f"{webserver_name_suffix}_interface"
+
+        publicip_info = self.create_publicip_address(publicip_name, group_name)
+
+        try:
+            self.create_nic(interface_name, group_name, vnet_name, 
+                            subnet_name, webserver_name_suffix, publicip_name)
+
+            logger.info(f"Creating {webserver_name_suffix} instance")
+            vm_info = self.compute_client.virtual_machines.begin_create_or_update(
+                group_name,
+                webserver_name_suffix,
+                create_server_params(group_name, webserver_name_suffix, interface_name, True)
+            ).result()
+        except ResourceNotFoundError as e:
+            logger.error(f"There may have been an error creating a webserver's dependent resource: {str(e)}")
+            SystemExit(1)
+
+        logger.info(f"{webserver_name_suffix} deployment status: {vm_info.provisioning_state}")
+
+        # Install and setup the webserver (IIS) feature
+        self.install_iis(group_name, webserver_name_suffix)
+
+        self.print_webserver_ip_address(webserver_name_suffix, publicip_info)    
+
+    def create_publicip_address(self, publicip_name, group_name):
+        """Create public IP address resource
+
+        Args:
+            publicip_name (str): Name of Public IP resource
+            group_name (str): Name of resource group
+
+        Returns:
+            JSON: JSON data of Public IP resource
+        """
+        # Likely Public IP Count Limit reached error
+        try:
+            publicip_info = self.net_client.public_ip_addresses.begin_create_or_update(
+                group_name,
+                publicip_name,
+                create_publicip_params()
+            ).result()
+            logger.info(f"{publicip_name} created successfully.")
+
+            return publicip_info
+        except HttpResponseError as e:
+            logger.error(f"Public IP count limit may have been reached: {str(e)}")
```

Comment:
This might lead to further errors down the line. Consider handling this more gracefully, perhaps by returning None and checking the return value in the calling code.